crytic_compile/crytic_compile.py

```diff
@@ -204,7 +204,10 @@ def __init__(self, target: Union[str, AbstractPlatform], **kwargs: str) -> None:
 
         self.libraries: Optional[Dict[str, int]] = _extract_libraries(kwargs.get("compile_libraries", None))  # type: ignore
 
-        self._compile(**kwargs)
+        self.compilation_kwargs = kwargs
+
+        if not kwargs.get("crytic_defer_compilation") == "true":
```

Comment:
Could this be a boolean instead of a magic string?

```suggestion
        if not kwargs.get("crytic_defer_compilation", False):
```